promising/promising_context.py

```
@@ -584,12 +584,12 @@ def _resolve_start_soon_default(self, start_soon_default: bool | Sentinel) -> bo
             f"but `{type(start_soon_default)}` was given instead"
         )
 
-    def _resolve_children_start_soon(self, children_start_soon: bool | Sentinel) -> bool | Sentinel:
-        if isinstance(children_start_soon, bool) or children_start_soon is NOT_SET:
+    def _resolve_children_start_soon(self, children_start_soon: bool | None | Sentinel) -> bool | None:
```

Ruff (0.15.5)

[warning] Boolean-typed positional argument in function definition

(FBT001)